### PR TITLE
Remove unnecessary `onColumnResize/Start/End` from useTableColumnResizeState

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-stately/table/src/useTableColumnResizeState.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ export function useTableColumnResizeState<T>(props: TableColumnResizeStateProps<
     getDefaultMinWidth,
     onColumnResizeStart: propsOnColumnResizeStart,
     onColumnResizeEnd: propsOnColumnResizeEnd,
+    onColumnResize: propsOnColumnResize,
```
Comment (LFDanLu) [inline packages/@react-stately/table/src/useTableColumnResizeState.ts:67]: I'm not entirely convinced we need this tbh, doesn't really seem to differ from `onResize` from useTableColumnResize

File: packages/@react-stately/table/src/useTableColumnResizeState.ts [hunk 2]
```diff
@@ -101,9 +102,9 @@ export function useTableColumnResizeState<T>(props: TableColumnResizeStateProps<
     let map = new Map(Array.from(uncontrolledColumns).map(([key]) => [key, newSizes.get(key)]));
     map.set(key, width);
     setUncontrolledWidths(map);
-
+    propsOnColumnResize(newSizes);
```
Comment (LFDanLu) [inline packages/@react-stately/table/src/useTableColumnResizeState.ts:105]:
```suggestion
    propsOnColumnResize?.(newSizes);
```

File: packages/@react-stately/table/src/useTableColumnResizeState.ts (hunk 1 above)
Comment (snowystinger) [reply packages/@react-stately/table/src/useTableColumnResizeState.ts:67]: Are we actually using `onColumnResizeStart` and `onColumnResizeEnd`? I think those two should go away in order to match TableLayout
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/layout/src/TableLayout.ts#L94

Comment (LFDanLu) [reply packages/@react-stately/table/src/useTableColumnResizeState.ts:67]: Technically unnecessary for the aria example, but `onColumnResizeStart` and `onColumnResizeEnd` provide different info from their `onResize` counterparts, namely they return the resize column key.

Comment (snowystinger) [reply packages/@react-stately/table/src/useTableColumnResizeState.ts:67]: Yes, but the key is only used by the stately/layout, I don't think anyone needs to have that, they already have the info from the stately object https://github.com/adobe/react-spectrum/blob/f7693f1cd7e0e44ab113d15e42d0777e1aa9c5e3/packages/%40react-stately/table/src/useTableColumnResizeState.ts#L119

**Review by snowystinger — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]

**Pull request merged**